torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -53,20 +67,31 @@ class SubgraphSlicer {
 
   void cleanupSubgraphs() {
     auto curNode = *block_->nodes().rbegin();
-    while (curNode != *block_->nodes().rend()) {
-      // Save the previous node, since we might delete `curNode` in next block
-      auto prevNode = curNode->prev();
-      if (curNode->kind() == prim::DifferentiableGraph) {
-        // Inlining nodes may cause some subexpression to come back in the
-        // subgraphs (for example, copying constants in repeatedly will generate
-        // redundant prim::Constants). Run CSE to clean them up.
-        EliminateCommonSubexpression(curNode->g(attr::Subgraph));
-
-        if (!inlineIfTooSmall(curNode)) {
-          diff_nodes_.push_back(curNode);
-        }
+
+    bool any_changed = true;
+
+    while(any_changed) {
```

Comment:
Can you move the any_changed loop inside unfuseAliasedOutputs? we have no reason to run the other passes in a loop